pkg/model/hetznermodel/firewall.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hetznermodel
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hetznercloud/hcloud-go/hcloud"
+	"k8s.io/kops/pkg/apis/kops"
+	"k8s.io/kops/upup/pkg/fi"
+	"k8s.io/kops/upup/pkg/fi/cloudup/hetzner"
+	"k8s.io/kops/upup/pkg/fi/cloudup/hetznertasks"
+)
+
+// ExternalAccessModelBuilder configures Firewall objects
+type ExternalAccessModelBuilder struct {
+	*HetznerModelContext
+	Lifecycle fi.Lifecycle
+}
+
+var _ fi.ModelBuilder = &ExternalAccessModelBuilder{}
+
+func (b *ExternalAccessModelBuilder) Build(c *fi.ModelBuilderContext) error {
+	//var sshAccess []net.IPNet
+	//for _, cidr := range b.Cluster.Spec.SSHAccess {
+	//	_, ipnet, err := net.ParseCIDR(cidr)
+	//	if err != nil {
+	//		return err
+	//	}
+	//	sshAccess = append(sshAccess, *ipnet)
+	//}
+	//var apiAccess []net.IPNet
+	//for _, cidr := range b.Cluster.Spec.KubernetesAPIAccess {
+	//	_, ipnet, err := net.ParseCIDR(cidr)
+	//	if err != nil {
+	//		return err
+	//	}
+	//	apiAccess = append(apiAccess, *ipnet)
+	//}
+
+	controlPlaneLabelSelector := []string{
+		fmt.Sprintf("%s=%s", hetzner.TagKubernetesClusterName, b.ClusterName()),
+		fmt.Sprintf("%s=%s", hetzner.TagKubernetesInstanceRole, string(kops.InstanceGroupRoleMaster)),
+	}
+	controlPlaneFirewall := &hetznertasks.Firewall{
+		Name:      fi.String("control-plane." + b.ClusterName()),
+		Lifecycle: b.Lifecycle,
+		Selector:  strings.Join(controlPlaneLabelSelector, ","),
+		Rules: []*hetznertasks.FirewallRule{
+			{
+				Direction: string(hcloud.FirewallRuleDirectionIn),
+				SourceIPs: b.Cluster.Spec.SSHAccess,
+				Protocol:  string(hcloud.FirewallRuleProtocolTCP),
+				Port:      fi.String("22"),
+			},
+		},
+		Labels: map[string]string{
+			hetzner.TagKubernetesClusterName:  b.ClusterName(),
+			hetzner.TagKubernetesFirewallRole: "control-plane",
+		},
+	}
+	if !b.UseLoadBalancerForAPI() {
```

Comment:
It would be cool to put this logic into `pkg/model` and have all the cloud providers share it!  But not a blocker